### PR TITLE
add default border radius to River visuals

### PR DESCRIPTION
**Pull request by rezrah**
> [!Important]  
> PR includes visual breaking changes. 
> Release tag: `next-minor`

🔗 [Read the rendered changeset for full details](https://github.com/primer/brand/blob/rezrah/add-river-rounded-corners/.changeset/silly-taxis-hammer.md) (To be shared with partner teams ahead of a potential release)

--- 

## Summary

Part of https://github.com/github/primer/issues/2533

Adds rounded corners to River visuals by default, makes shadow opt-in and renames Storybook subfolder for River stories from fixtures to features for consistency.

These changes bring the River.Visual up-to-date with the latest site refactor expectations.

As this will be a visual breaking change, this will first be merged into the `next-minor` branch (when available) and shared with partner teams to gauge impact ahead of a GA release.

🔗  [Try it in Storybook](https://primer-775d7eed63-26139705.drafts.github.io/storybook/?path=/story/components-river--default)

## What should reviewers focus on?

- Spotting bugs or regressions in other components / layouts that use River
- Code and design review

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2023-08-07 at 11 34 53

 </td>
<td valign="top">

Screenshot 2023-08-07 at 11 35 55

</td>
</tr>
</table>

### CODE REVIEW

File: packages/react/src/Avatar/Avatar.visual.spec.ts-snapshots/Visual-Comparison-Avatar-Avatar-Default-1-linux.png
Comment (danielguillan) [inline packages/react/src/Avatar/Avatar.visual.spec.ts-snapshots/Visual-Comparison-Avatar-Avatar-Default-1-linux.png:1]: Any idea why this snapshot changed?

**Review by danielguillan — APPROVED** [review]

File: packages/react/src/Avatar/Avatar.visual.spec.ts-snapshots/Visual-Comparison-Avatar-Avatar-Default-1-linux.png
Comment (rezrah) [reply packages/react/src/Avatar/Avatar.visual.spec.ts-snapshots/Visual-Comparison-Avatar-Avatar-Default-1-linux.png:1]: Good question, this is a flakey test. The current version of the snapshot didn't load the avatar in time, so the new one is adding it back it and reverting it to how it should be. This is why everyone was seeing the 'visual changes found' message in their PR's. This should stop that happening now 🤞.

File: packages/react/src/River/River.module.css [hunk 1]
```diff
@@ -22,6 +22,11 @@
   object-fit: cover;
 }
 
+.River__visual--rounded,
+.River__visual--rounded img {
+  border-radius: var(--brand-borderRadius-large);
+}
```
Comment (josepmartins) [inline packages/react/src/River/River.module.css:28]: we might need to add video as well, the posters could take full width and height of the visual area.

```suggestion
.River__visual--rounded,
.River__visual--rounded video,
.River__visual--rounded img {
  border-radius: var(--brand-borderRadius-large);
}
```

Comment (josepmartins) [inline packages/react/src/River/River.module.css:28]: another would be to apply it to the visual area + use overflow hidden

```suggestion
.River__visual--rounded {
  overflow: hidden;
  border-radius: var(--brand-borderRadius-large);
}
```

Comment (rezrah) [reply packages/react/src/River/River.module.css:28]: I like this suggestion, handles more use-cases, while separating concerns in the parent 👍. Going with this approach.

**Review by jesussandreas — APPROVED** [review]

**Pull request merged**